src/Uno.UI/UI/Xaml/VisualStateGroup.cs

```diff
@@ -335,7 +336,23 @@ void ApplyTargetStateSetters()
 
 					while (settersEnumerator.MoveNext())
 					{
-						settersEnumerator.Current.ApplyValue(DependencyPropertyValuePrecedences.Animations, element);
+						var current = settersEnumerator.Current;
+
+						// todo: add why we are doing this
+						if (current is IDependencyObjectStoreProvider provider)
+						{
+							var expression = provider.Store.GetBindingExpression(Setter.ValueBackingProperty);
+							if (expression is { } && expression.ParentBinding is { RelativeSource.Mode: RelativeSourceMode.TemplatedParent, Converter: { } })
+							{
+								// force binding value to re-evaluate
+								expression.SuspendBinding();
+								expression.ResumeBinding();
```

Comment:
We'll probably need to find a better way to do this. Did `expression.ApplyBinding()` do the work?